discovery/sync_manager.go

```diff
@@ -200,8 +201,9 @@ type SyncManager struct {
 	// number of queries.
 	rateLimiter *rate.Limiter
 
-	wg   sync.WaitGroup
-	quit chan struct{}
+	wg     sync.WaitGroup
+	quit   chan struct{}
```

Comment:
If the goal is to rely on a single ctx to start and shut down all the subservices, should we rely on the `cancel` instead of `quit`?

Reply:
we need both. 

`quit` is about stopping a systems internal functions and controlled wholly from within the system. `cancel` is about canceling any calls we make that are going out of the system to other systems. When calls come _in from other subsystems_ that dont take our own context: then we listen on that caller's ctx along with our own quit. Otherwise, we'd need to pass around our ctx everywhere (we'd need to put it in a struct which is an anti-pattern) and then listen on both the incoming caller's ctx and our own. 
I think it is much cleaner to just have our calling ctx and our internal quit as separate things.

Reply:
>  I think it is much cleaner to just have our calling ctx and our internal quit as separate things.

> It will be aborted because if it's ctx is being cancelled, then quit will also be closed. we can listen on both - but it is redundant

I think these two statements contradict each other. If `quit` is identical to `ctx`, then there's no need to have two sources of truth. If `ctx` is planned to be different than `quit`, then whenever we listen for `quit`, we check for ctx done since one's internal and the other can be external?